injector/cpu_pressure.go

```diff
@@ -141,8 +143,20 @@ func (i cpuPressureInjector) Inject() error {
 		}(core)
 	}
 
+	waitCh := make(chan struct{})
+
+	go func() {
+		wg.Wait()
+		close(waitCh)
+	}()
+
 	// wait for stress routines to finish initializing
-	wg.Wait()
+	select {
+	case <-ctx.Done():
```

Comment:
Here we will exit either once all the stressors are initialized or once we're cancelled. I could also add a check in the loop where we create initializers